src/sgame/sg_admin.cpp

```diff
@@ -5579,135 +5579,128 @@ void G_admin_cleanup()
 	g_admin_commands = nullptr;
 }
 
-bool G_admin_bot( gentity_t *ent )
-{
-	int min_args = 3;
-	char arg1[MAX_TOKEN_CHARS];
-	char name[MAX_TOKEN_CHARS];
-	char team[MAX_TOKEN_CHARS];
-	char skill[MAX_TOKEN_CHARS];
-	char err[MAX_STRING_CHARS];
-	char behavior[MAX_QPATH];
-	int skill_int;
-	int i;
-	int clientNum;
+static void BotUsage( gentity_t *ent ) {
+	static const char bot_usage[] = QQ( N_( "^3bot: ^7usage: bot add (<name> | *) (aliens | humans) [<skill level> [<behavior>]]\n"
+	                                        "            bot del (<name> | all)\n"
+	                                        "            bot names (aliens | humans) <names>…\n"
+	                                        "            bot names (clear | list)" ) );
+	ADMP( bot_usage );
+}
 
-	static const char bot_usage[] = QQ( N_( "^3bot: ^7usage: bot add [^5name|*^7] [^5aliens|humans^7] (^5skill^7) (^5behavior^7)\n"
-	                                        "            bot del [^5name|all^7]\n"
-	                                        "            bot names [^5aliens|humans^7] [^5names…^7]\n"
-	                                        "            bot names [^5clear|list^7]" ) );
+static team_t TeamFromString2( const char* s ) {
+	// exact match only
+	if ( !Q_stricmp( s, "a" ) || !Q_stricmp( s, "aliens" ) )
+	{
+		return team_t::TEAM_ALIENS;
+	}
+	if ( !Q_stricmp( s, "h" ) || !Q_stricmp( s, "humans" ) )
+	{
+		return team_t::TEAM_HUMANS;
+	}
+	return team_t::TEAM_NONE;
+}
 
-	if ( trap_Argc() < min_args )
+static bool BotAddCmd( gentity_t *ent, const Cmd::Args& args ) {
+	if ( args.Argc() < 4 || args.Argc() > 6 )
 	{
-		ADMP( bot_usage );
+		BotUsage( ent );
 		return false;
 	}
-	trap_Argv( 1, arg1, sizeof( arg1 ) );
-	trap_Argv( 2, name, sizeof( name ) );
 
-	if ( !Q_stricmp( arg1, "add" ) )
+	const char* name = args[2].data();
+	if ( !Q_stricmp( name, "all" ) )
 	{
-		RETURN_IF_INTERMISSION;
+		ADMP( QQ( N_( "bots can't have that name" ) ) );
+		return false;
+	}
 
-		min_args++; //now we also need a team name
-		if ( !Q_stricmp( name, "all" ) )
-		{
-			ADMP( QQ( N_( "bots can't have that name" ) ) );
-			return false;
-		}
-		if ( trap_Argc() < min_args )
-		{
-			ADMP( bot_usage );
-			return false;
-		}
-		trap_Argv( 3, team, sizeof( team ) );
+	team_t team = TeamFromString2( args[3].data() );
+	if ( team == team_t::TEAM_NONE )
+	{
+		ADMP( QQ( N_( "Invalid team name" ) ) );
+		BotUsage( ent );
+		return false;
+	}
 
-		//skill level checks
-		min_args++;
-		if ( trap_Argc() < min_args )
-		{
-			skill_int = 5; //no skill arg
-		}
-		else
-		{
-			trap_Argv( 4, skill, sizeof( skill ) );
-			skill_int = atoi( skill );
-			if ( skill_int < 1 )
-			{
-				skill_int = 1; //skill arg too small, reset to 1
-			}
-			else if ( skill_int > 10 )
-			{
-				skill_int = 10; //skill arg too bit, reset to 10
-			}
-		}
+	int skill;
+	if ( args.Argc() >= 5 )
+	{
+		skill = Math::Clamp( atoi( args[4].data() ), 1, 10 );
```

Comment:
Perhaps print a warning to the user letting them know that the skill they entered was incorrect?